packages/template-block/src/components/CustomButton.tsx

```diff
@@ -0,0 +1,55 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Settings } from '../types';
+import { BlockStyles, RichTextEditor, convertToRteValue, hasRichTextValue } from '@frontify/guideline-blocks-settings';
+import { PluginComposer } from '@frontify/fondue';
+import { useRef, useState } from 'react';
+
+export type CustomButtonProps = {
+    blockSettings: Settings;
+    isEditing: boolean;
+    isDisabled: boolean;
+    updateBlockSettings: (newSettings: Partial<Settings>) => Promise<void>;
+    handleNewPublication: () => void;
+};
+
+export const CustomButton = ({
+    blockSettings,
+    isEditing,
+    isDisabled,
+    updateBlockSettings,
+    handleNewPublication,
+}: CustomButtonProps) => {
+    const screenReaderRef = useRef<HTMLDivElement>(null);
+    const [buttonHover, setButtonHover] = useState<boolean>(false);
+    const { buttonText } = blockSettings;
+
+    return (
+        <button
+            data-test-id="template-block-new-publication-btn"
+            disabled={isDisabled}
+            onClick={isEditing ? undefined : handleNewPublication}
+            onMouseEnter={() => setButtonHover(true)}
+            onMouseLeave={() => setButtonHover(false)}
+            style={{
+                ...BlockStyles.buttonSecondary,
+                ...(buttonHover ? BlockStyles.buttonSecondary?.hover : null),
+                ...(isDisabled ? { opacity: 0.5 } : null),
+            }}
+        >
+            <RichTextEditor
```

Comment:
Is the RichTextEditor here needed as it is used without any toolbar and plugin?

Reply:
Basically, it's a foundation for a future button customisation improvements.